src/com/textFinder/TextFinderImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.textFinder;
+
+import java.io.File;
+import java.util.*;
+import java.util.concurrent.*;
+
+import com.textFinder.finder.FileSearchThreadRunner;
+
+public class TextFinderImpl implements TextFinder {
+
+    @Override
+    public List<File> findText(File file, String searchText) {
+        File absoluteFile = file.getAbsoluteFile();
+
+        List<File> result = Collections.synchronizedList(new ArrayList<>());
+        ExecutorService executor = Executors.newCachedThreadPool();
+        List<Future<?>> futures = new ArrayList<>();
+
+        searchFiles(absoluteFile, searchText, result, executor, futures);
+        for (Future<?> future : futures) {
+            try {
+                future.get();
+            } catch (InterruptedException | ExecutionException e) {
+                System.err.println("Error during file search: " + e.getMessage());
+                e.printStackTrace();
+            }
+        }
+
+        executor.shutdown();
```

Comment:
don't shut down here. Instead implement `Closable` interface and move the creation of this class in try-with-resources:

```java
try (TextFinder textFinder = new TextFinderImpl()) {
...
}
```

Ref: https://docs.oracle.com/javase/8/docs/api/java/io/Closeable.html